src/extension_commands.ml

```diff
@@ -96,6 +96,92 @@ let _upgrade_ocaml_lsp_server =
+      let sandbox = Extension_instance.sandbox instance in
+      match sandbox with
+      | Dune dune ->
+        let* is_dune_locked = Dune.detect_dune_lock_dir dune in
+        if is_dune_locked
+        then
+          let* dune_lsp_present = Dune.detect_dune_ocamllsp dune in
+          if dune_lsp_present
+          then
+            show_message `Info "OCaml-LSP server is already installed." |> Promise.return
+          else (
+            let options =
+              ProgressOptions.create
+                ~location:(`ProgressLocation Notification)
+                ~title:"Running `dune tools exec ocamllsp`"
+                ~cancellable:false
+                ()
+            in
+            let task ~progress:_ ~token:_ =
+              let+ result =
+                (* We first check the version so that the process can exit, otherwise the progress indicator runs \
+                   forever as the dune tools exec command automatically starts the server hence the process never ends.*)
+                Dune.exec dune ~args:[ "tools"; "exec"; "ocamllsp"; "--version" ]
+                |> Cmd.output ~cwd:(Dune.root dune)
+              in
+              match result with
+              | Ok _ -> Ojs.null
+              | Error err ->
+                show_message
+                  `Error
+                  "An error occured while running `dune tools exec ocamllsp`: %s"
+                  err;
+                Ojs.null
+            in
+            let+ _ = Vscode.Window.withProgress (module Ojs) ~options ~task in
+            let _ =
+              Dune.exec dune ~args:[ "tools"; "exec"; "ocamllsp" ]
+              |> Cmd.output ~cwd:(Dune.root dune)
+            in
+            let _ = Extension_instance.start_language_server instance in
+            ())
+        else Extension_instance.suggest_to_run_dune_pkg_lock () |> Promise.return
+      | _ -> Promise.return ()
+    in
+    ()
+  in
+  command Extension_consts.Commands.install_dune_lsp handler
+;;
+
+let _run_dune_pkg_lock =
+  let handler (instance : Extension_instance.t) ~args:_ =
+    let open Promise.Syntax in
+    let _ =
+      match Extension_instance.sandbox instance with
+      | Dune dune ->
+        let options =
+          ProgressOptions.create
+            ~location:(`ProgressLocation Notification)
+            ~title:"Running `dune pkg lock`"
+            ~cancellable:false
+            ()
+        in
+        let task ~progress:_ ~token:_ =
+          let+ result =
+            Dune.exec dune ~args:[ "pkg"; "lock" ] |> Cmd.output ~cwd:(Dune.root dune)
+          in
+          match result with
+          | Ok _ -> Ojs.null
+          | Error err ->
+            show_message `Error "An error occured while running dune pkg lock: %s" err;
+            Ojs.null
+        in
+        let+ _ = Vscode.Window.withProgress (module Ojs) ~options ~task in
+        let _ = Extension_instance.start_language_server instance in
+        ()
+      | _ -> Promise.return ()
```

Comment:
Same as above: I'd log something here to facilitate debugging if this case ends up being hit.